test/e2e/servicedns.go

```diff
@@ -56,6 +59,7 @@ var _ = Describe("ServiceDNS", func() {
 	BeforeEach(func() {
 		fedClient = f.FedClient(userAgent)
 		namespace = f.TestNamespaceName()
+		domainClient = fedClient.MulticlusterdnsV1alpha1().Domains(f.FederationSystemNamespace())
```

Comment:
Please CMIIW, here the `namespace` is pointing to test namespace. The Domain object is intended to be created in federation system namespace and multiple `ServiceDNSRecord` objects can refer to it.

Reply:
The Domain object needs to created in the federation system namespace because we refer to it in ServiceDNSRecord by just the name (`DomainRef`). It is assumed that these Domain Objects are system-wide entities and not contained by namespace (similar to `FederatedCluster` objects). The intention is that an `Federation Admin` would create this Domain objects and an `App Admin` would create the `ServiceDNSRecords`.